x/farming/keeper/proposal_handler_test.go

```diff
@@ -406,80 +406,120 @@ func (suite *KeeperTestSuite) TestValidateDeletePublicPlanProposal() {
+				plan.GetStartTime(),
+				plan.GetEndTime(),
+				sdk.NewCoins(sdk.NewInt64Coin("stake", 100_000)),
+				sdk.ZeroDec(),
+			),
+		}, nil),
 	)
 	suite.Require().NoError(err)
 
 	plan, found = suite.keeper.GetPlan(suite.ctx, uint64(1))
 	suite.Require().Equal(true, found)
 	suite.Require().Equal(plan.(*types.FixedAmountPlan).EpochAmount, sdk.NewCoins(sdk.NewInt64Coin("stake", 100_000)))
 
-	// update the fixed amount plan back to ratio plan
-	updateRequests = []*types.UpdateRequestProposal{
-		types.NewUpdateRequestProposal(
-			plan.GetId(),
-			plan.GetName(),
-			plan.GetFarmingPoolAddress().String(),
-			plan.GetTerminationAddress().String(),
-			plan.GetStakingCoinWeights(),
-			plan.GetStartTime(),
-			plan.GetEndTime(),
-			nil,
-			sdk.NewDecWithPrec(7, 2), // 7%
-		)}
-
+	// update back to ratio plan with different epoch ratio
 	err = keeper.HandlePublicPlanProposal(
 		suite.ctx,
 		suite.keeper,
-		types.NewPublicPlanProposal("testTitle", "testDescription", nil, updateRequests, nil),
+		types.NewPublicPlanProposal("testTitle", "testDescription", nil, []*types.UpdateRequestProposal{
+			types.NewUpdateRequestProposal(
+				plan.GetId(),
+				plan.GetName(),
+				plan.GetFarmingPoolAddress().String(),
+				plan.GetTerminationAddress().String(),
+				plan.GetStakingCoinWeights(),
+				plan.GetStartTime(),
+				plan.GetEndTime(),
+				nil,
+				sdk.NewDecWithPrec(7, 2), // 7%
+			),
+		}, nil),
 	)
 	suite.Require().NoError(err)
 
 	plan, found = suite.keeper.GetPlan(suite.ctx, uint64(1))
 	suite.Require().Equal(true, found)
 	suite.Require().Equal(plan.(*types.RatioPlan).EpochRatio, sdk.NewDecWithPrec(7, 2))
 }
+
+func (suite *KeeperTestSuite) TestDeletePublicPlan() {
+	// create a fixed amount public plan
+	err := keeper.HandlePublicPlanProposal(
+		suite.ctx,
+		suite.keeper,
+		types.NewPublicPlanProposal("testTitle", "testDescription", []*types.AddRequestProposal{
+			types.NewAddRequestProposal(
+				"testPlan",
+				suite.addrs[0].String(),
+				suite.addrs[1].String(),
+				sdk.NewDecCoins(
+					sdk.NewDecCoinFromDec(denom1, sdk.NewDecWithPrec(3, 1)),
+					sdk.NewDecCoinFromDec(denom2, sdk.NewDecWithPrec(7, 1)),
+				),
+				types.ParseTime("0001-01-01T00:00:00Z"),
+				types.ParseTime("9999-12-31T00:00:00Z"),
+				sdk.NewCoins(sdk.NewInt64Coin(denom3, 100_000_000)),
+				sdk.ZeroDec(),
+			),
+		}, nil, nil),
+	)
+	suite.Require().NoError(err)
+
+	plan, found := suite.keeper.GetPlan(suite.ctx, uint64(1))
+	suite.Require().Equal(true, found)
+
+	// check farming pool address initial balances
+	poolInitialBalances := suite.app.BankKeeper.GetAllBalances(suite.ctx, suite.addrs[0])
+	suite.Require().Equal(initialBalances, poolInitialBalances)
+
+	// delete the plan
+	err = keeper.HandlePublicPlanProposal(
+		suite.ctx,
+		suite.keeper,
+		types.NewPublicPlanProposal("testTitle", "testDescription", nil, nil, []*types.DeleteRequestProposal{
+			types.NewDeleteRequestProposal(plan.GetId()),
+		}),
+	)
```

Comment:
Could you add a test case to asserting even if a plan has the same `TerminationAddress` with `FarmingPoolAddress` the plan set terminated

Reply:
I can add the test case to cover when `TerminationAddress` with `FarmingPoolAddress` are the same. However, as plan gets removed from the store, I am not sure if there is a way to check whether or not the Terminated is set to true.

Reply:
Yes, we can't check the plan state after deletion, but we can check the successfully removed with balance differences using `farming.EndBlocker(suite.ctx, suite.keeper)` on test code after handle `PublicPlanProposal` and also events like https://github.com/cosmos/cosmos-sdk/blob/d60cb3a49f73c8f51da80076d22d3553c5a8b20b/x/bank/keeper/keeper_test.go#L1087-L1113